jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -4273,8 +4273,50 @@ public boolean matches(String expr) {
 	 *
 	 * @since 1.4
 	 */
-	public String replaceAll(String expr, String substitute) {
-		return Pattern.compile(expr).matcher(this).replaceAll(substitute);
+	public String replaceAll(String regex, String substitute) {
+		if (regex != null && substitute != null && regex.lengthInternal() == 1 && !hasMetaChars(regex)) {
+			int substituteLength = substitute.lengthInternal();
+			int length = lengthInternal();
+			int newCharIndex = 0;
+			if (substituteLength < 2) {
+				if (enableCompression && isCompressed() && (substituteLength == 0 || substitute.isCompressed())) {
+					/*[IF Sidecar19-SE]*/
+					byte[] newChars = new byte[length];
+					/*[ELSE]*/
+					char[] newChars = new char[(length + 1) >> 1];
+					/*[ENDIF]*/
+					byte toReplace = helpers.getByteFromArrayByIndex(regex, 0);
+					byte replacement = substituteLength == 1 ? helpers.getByteFromArrayByIndex(substitute, 0) : (byte)-1;
+					for (int i = 0; i < length; ++i) {
+						byte current = helpers.getByteFromArrayByIndex(value, i);
+						if (current != toReplace) {
+							helpers.putByteInArrayByIndex(newChars, newCharIndex++, current);
+						} else if (substituteLength == 1) {
+							helpers.putByteInArrayByIndex(newChars, newCharIndex++, replacement);
+						}
+					}
+					return new String(newChars, 0, newCharIndex, true);
+				} else if (!enableCompression || !isCompressed()) {
+					/*[IF Sidecar19-SE]*/
+					byte[] newChars = new byte[length << 1];
+					/*[ELSE]*/
+					char[] newChars = new char[length];
+					/*[ENDIF]*/
+					char toReplace = regex.charAtInternal(0);
+					char replacement = substituteLength == 1 ? substitute.charAtInternal(0) : (char)-1; 
```

Comment:
Same concern about `-1` here as well.

Reply:
same reasoning as above to say the -1 is never used/seen